packages/web-app/src/modules/earn-views/pages/EarningSummaryPage.tsx

```diff
@@ -58,6 +60,12 @@ const _EarningSummaryPage: FC<Props> = ({
   viewLast7Days,
   viewLast30Days,
 }) => {
+  const [selectedMachine, setSelectedMachine] = useState<MachineState | null>(null)
```

Comment:
let's use `id` instead of entire machine state object to avoid data duplication
```suggestion
  const [selectedMachineId, setSelectedMachineId] = useState<string | null>(null)
```